mne/preprocessing/_peak_finder.py

```diff
@@ -50,11 +50,26 @@ def peak_finder(x0, thresh=None, extrema=1, verbose=None):
     s = x0.size
 
     if x0.ndim >= 2 or s == 0:
-        raise ValueError('The input data must be a non empty 1D vector')
+        raise ValueError('The input data must be a non-empty 1D vector')
+
+    try:
+        eps = 1e2 * np.finfo(x0.dtype).eps
+    except ValueError:  # int dtypes are exact
+        eps = 0.
```

Comment:
I don't like try except blocks. They are too implicit. Here i would say that either we support integer inputs and you should check that dtype is int-like or if not you should cast to a float with the same nbytes. See

https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/validation.py#L173

how it's done in sklearn.

Reply:
I'm now simply casting unless it's already a float, see 20361b4

You okay with that?